prow/config/branch_protection.go

```diff
@@ -597,14 +600,23 @@ func (c *Config) BranchProtectionWarnings(logger *logrus.Entry, presubmits map[s
 //   - contexts that are always required to be present
 //   - contexts that are required, _if_ present
 //   - contexts that are always optional
-func BranchRequirements(branch string, jobs []Presubmit) ([]string, []string, []string) {
+func BranchRequirements(branch string, jobs []Presubmit, requiredNonOptionalJobs *bool) ([]string, []string, []string) {
 	var required, requiredIfPresent, optional []string
+	var nonOptionalJobs bool
+	if requiredNonOptionalJobs != nil && *requiredNonOptionalJobs {
+		nonOptionalJobs = true
+	}
 	for _, j := range jobs {
 		if !j.CouldRun(branch) {
 			continue
 		}
-
 		if j.ContextRequired() {
+			if nonOptionalJobs && j.RequiresManualTrigger() {
```

Comment:
Let's turn L614-L628 into a `switch` (with two cases and a `default`), it'll improve the structure here.